File: app-service-comments/src/main/java/com/github/arhor/aws/graphql/federation/comments/data/entity/UserRepresentationEntity.java
package com.github.arhor.aws.graphql.federation.comments.data.entity;

import lombok.Builder;
import lombok.experimental.FieldNameConstants;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Immutable;
import org.springframework.data.domain.Persistable;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.util.UUID;

@Table("user_representations")
@Immutable
@Builder(toBuilder = true)
@FieldNameConstants(asEnum = true)
public record UserRepresentationEntity(
    @Id
    @Column("id")
    UUID id
) implements Persistable<UUID> {

    @Override
    public UUID getId() {
        return id;
    }

    @Override
    public boolean isNew() {
        return true;
    }
}

File: app-service-posts/src/main/kotlin/com/github/arhor/aws/graphql/federation/posts/data/entity/UserRepresentationEntity.kt
package com.github.arhor.aws.graphql.federation.posts.data.entity

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Immutable
import org.springframework.data.domain.Persistable
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.util.UUID

@Table(UserRepresentationEntity.TABLE_NAME)
@Immutable
data class UserRepresentationEntity(
    @Id
    @Column("id")
    private val id: UUID,
) : Persistable<UUID> {

    override fun getId(): UUID = id

    override fun isNew(): Boolean = true

    companion object {
        const val TABLE_NAME = "user_representations"
    }
}
